ggml.h

```diff
@@ -773,6 +773,19 @@ int ggml_cpu_has_blas(void);
 int ggml_cpu_has_sse3(void);
 int ggml_cpu_has_vsx(void);
 
+//
+// threading for non posix systems
+//
+#ifndef _POSIX_THREADS
+#if defined(_WIN32)
+#include <windows.h>
+#endif
+typedef HANDLE pthread_t;
+typedef DWORD thread_ret_t;
+static int pthread_create(pthread_t* out, void* unused, thread_ret_t(*func)(void*), void* arg);
+static int pthread_join(pthread_t thread, void* unused);
+#endif
```

Comment:
Why reinvent `std::thread`?

Reply:
std::thread does not exist in msvc

Reply:
Seems to exist at https://learn.microsoft.com/en-us/cpp/standard-library/thread-class?view=msvc-170  . But I didn't test, so feel free to resolve this thread if `std::thread` doesn't work on your machine